addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -84,13 +87,72 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, fmt.Errorf("failed to enable mirroring the storagecluster %q in namespace %q in managed cluster. Error %v", scr.Name, scr.Namespace, err)
 	}
 
+	err = r.createS3(ctx)
+	if err != nil {
+		klog.Error(err, "Failed to create ODR S3 resources")
+		return ctrl.Result{}, err
+	}
+
 	err = r.createVolumeReplicationClass(ctx, &mirrorPeer)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to create volumereplicationclass: %v", err)
 	}
 	return ctrl.Result{}, nil
 }
 
+func (r *MirrorPeerReconciler) createS3(ctx context.Context) error {
+	var err error
+	bucket := "odrbucket"
+	namespace := "openshift-dr-system"
+	if newNamespace := os.Getenv("ODR_NAMESPACE"); newNamespace != "" {
+		namespace = newNamespace
+	}
+
+	noobaaOBCNamespace := &corev1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "namespace",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	}
+	err = r.SpokeClient.Get(ctx, types.NamespacedName{Name: namespace}, noobaaOBCNamespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.Info("Could not find ODR OBC namespace")
+			return err
+		} else {
+			klog.Error(err, "Failed to get OBC namespace")
+			return err
+		}
+	}
```

Comment:
Why you are fetching namespace? If a namespace is not found OBC creation will fail automatically.

Reply:
I kept it for the message to prompt the user, but I think the error message from the OBC getter should suffice, I'll remove this.